tests/test_training_simple.py

```diff
@@ -0,0 +1,16 @@
+from open_lm.main import main
+
+
+def test_train_simple():
+    seq_len = 2048
+    num_batches = 5
+    batch_size = 1
+    # fmt: off
+    main([
+        "--train-num-samples", str(num_batches * seq_len),
+        "--batch-size", str(batch_size),
+        "--dataset-type", "synthetic",
+        "--model", "open_lm_11m",
+        "--epochs", "1",
+    ])
+    # fmt: on
```

Comment:
probably need something like:
`success = main(main_args)`
`assert success != -1`

Reply:
main() should throw an exception if anything breaks, which will cause the test to fail. This is probably ideal for  tests, running from command line, and for development (so you don't need to think about returning 0/1 from main and just throw an exception instead). What do you think?

Reply:
agreed, though right now main returns -1 in many failure cases. so we can either add throwing to this PR or check for -1?

Reply:
oh, didn't even realize we did that. I'll add throwing to this PR.

Reply:
added! changed all `return -1` to `raise ValueError` instead

Reply:
nice! lgtm!